### ISSUE TITLE
latest symlink failing to create

### ISSUE THREAD

**Issue by tim-oe** [offen/docker-volume-backup#509]
**Describe the bug**
failing to create symlink error log
level=ERROR msg="Unexpected error running schedule 40 18 * * *: input/output error" error="main.runScript.func4: error runnin                        g script: main.(*script).copyArchive: error copying archive: local.(*localStorage).Copy: error creating latest symlink: symlink backup-2024-12-17T18-40-00.tar.gz /archive/backup-latest.tar.gz: input/output error"

**To Reproduce**
ran chron based backup

**Expected behavior**
latest symlink is created

**Version (please complete the following information):**
 - Image Version: v2.43.0/latest
 - Docker Version: 27.4.0, build bde2b89
 - Docker Compose Version (if applicable): docker-compose-plugin/noble,now 2.31.0-1~ubuntu.24.04~noble amd64

**Additional context**
docker-compose,yml:

---
services:
  vaultwarden:
    image: vaultwarden/server:latest
    container_name: vaultwarden
    hostname: vaultwarden
    restart: unless-stopped
    ports:
     - 8860:80
    volumes:
      - vaultwarden-storage:/data:rw
    environment:
     - ROCKET_PORT=8860
     - ADMIN_TOKEN=${ADMIN_TOKEN}
     - WEBSOCKET_ENABLED=true
     - SIGNUPS_ALLOWED=false
     - SMTP_HOST=${SMTP_HOST}
     - SMTP_FROM=${SMTP_FROM}
     - SMTP_PORT=${SMTP_PORT}
     - SMTP_SSL=${SMTP_SSL}
     - SMTP_USERNAME=${SMTP_USERNAME}
     - SMTP_PASSWORD=${SMTP_PASSWORD}
     - DOMAIN=${DOMAIN}
    labels:
      # for offen backup
      - docker-volume-backup.stop-during-backup=true
  backup:
    # In production, it is advised to lock your image tag to a proper
    # release version instead of using `latest`.
    # Check https://github.com/offen/docker-volume-backup/releases
    # for a list of available releases.
    image: offen/docker-volume-backup:latest
    container_name: vaultwarden-backup
    restart: always
    environment:
      BACKUP_CRON_EXPRESSION: "40 18 * * *"
      BACKUP_FILENAME: backup-%Y-%m-%dT%H-%M-%S.tar.gz
      BACKUP_LATEST_SYMLINK: backup-latest.tar.gz
      BACKUP_PRUNING_PREFIX: backup-
      BACKUP_RETENTION_DAYS: 7
    volumes:
      # force local TZ
      - /etc/timezone:/etc/timezone:ro
      - /etc/localtime:/etc/localtime:ro
      - vaultwarden-storage:/backup/my-app-backup:ro
      - /var/run/docker.sock:/var/run/docker.sock:ro
      - /media/docker_backup/vaultwarden:/archive
volumes:
  vaultwarden-storage:
    name: vaultwarden-storage
networks:
  default:
    external: true
    name: share-net

**Comment by m90** [comment]
What kind of filesystem is your `/media` mount using and how is it mounted?

**Comment by m90** [comment]
#429 is possibly related

**Comment by m90** [comment]
In any case, it's unlikely this is something that can be dealt with at the level of this tool. Creating a symlink is just doing a syscall, and in your scenario this syscall fails with [`EIO`](https://www.gnu.org/software/libc/manual/html_node/Error-Codes.html):

> Macro: int EIO
> “Input/output error.” Usually used for physical read or write errors.

My guess your issue is some combination of your OS not handling the mount of your external target correctly.

**Comment by tim-oe** [comment]
yes you are correct, i verified that i cannot manually create the link.
realized that this is a cifs mount to a nas so it's makes sense.
sorry for the false error.
